plugins/scaffolder-node/api-report-alpha.md

```diff
@@ -19,10 +19,11 @@ export type AutocompleteHandler = ({
 }: {
   resource: string;
   token: string;
-  context: Record<string, string>;
+  context: Record<string, string | undefined>;
 }) => Promise<{
   results: {
     title: string;
+    context?: Record<string, unknown>;
```

Comment:
What's the motivation for adding a context on this return here? What's the usecase and what do we want to store in this context?

Reply:
Hi @benjdlambert , I added a new property named `context` because there may be use cases where I not only need the `title` but also other props. For example, in the Gitlab use case, after fetching first the available groups, other than the group names I also need their ids, so that I can do a next query by fetching repos based on those ids. Should we change the variable name?

Reply:
I wonder if it just makes sense to include id instead rather than having an unconstrained type here that's pretty ambiguous? Thinking that we can make `id` required, but go ahead and update the other resolvers too to return the ID. Maybe the ID should be the value that is actually used in the picker too, rather than the title?

Reply:
I was thinking about leaving it unconstrained because there may be other use cases where we would need other than the id maybe? Maybe `context` is not the right key name. 

So you are suggesting to only return `id`? What happens if for example only the title is needed? we Would need to call another api to get it?